src/server/replica.cc

```diff
@@ -625,16 +650,21 @@ error_code Replica::SendNextPhaseRequest(string_view kind) {
   return std::error_code{};
 }
 
-error_code DflyShardReplica::StartFullSyncFlow(BlockingCounter sb, Context* cntx) {
+std::error_code DflyShardReplica::StartSyncFlow(BlockingCounter sb, Context* cntx,
+                                                std::vector<LSN>& lsns, bool& is_full_sync) {
```

Comment:
Wow, so some shards can perform a full sync while others do a partial sync? And what about the full_sync_cut coutner?

Reply:
Ah got it, you actually check it below. But it would've been a cool feature 😎

Reply:
The full_sync_cut counter is decremented even in partial sync mode after all the entries have been submitted. The real problem with mixed mode sync is flushing the correct data when the master and the replica have different shard numbers :/